File: backend/app/schemas/draft.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict

from app.models.draft_section import DraftSectionType


class CreateDraftRequest(BaseModel):
    project_id: str
    title: str
    topic: str


class DraftSectionRequest(BaseModel):
    section: DraftSectionType
    content: str


class DraftSectionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    section: DraftSectionType
    content: str
    display_order: int


class DraftResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    title: str
    topic: str
    created_at: datetime
    updated_at: datetime


class DraftDetailResponse(DraftResponse):
    sections: list[DraftSectionResponse]

File: Client/paper-mind/src/types/draft.ts
export interface Draft {
  id: string;
  project_id: string;
  title: string;
  topic: string;

  paper_ids: string[];
  instructions?: string;

  created_at: string;
  updated_at: string;
}

export interface DraftSection {
  section: string;
  content: string;
}

export interface DraftDetail extends Draft {
  sections: DraftSection[];
}

export interface FullPaper {
  id: string;
  project_id: string;
  title: string;
  topic: string;
  content: string;
}

export interface CreateDraftRequest {
  project_id: string;
  title: string;
  topic: string;
  paper_ids: string[];
  instructions?: string;
}

export interface UpdatePaperRequest {
  content: string;
}